dbt/include/impala/macros/adapters.sql

```diff
@@ -190,24 +190,45 @@
 {% endmacro %}
 
 {% macro impala__drop_relation(relation) -%}
+  {% set rel_type = get_relation_type(relation) %}
   {% call statement('drop_relation', auto_begin=False) -%}
-    drop {{ relation.type }} if exists {{ relation }}
+    drop {{ rel_type }} if exists {{ relation }}
   {%- endcall %}
 {% endmacro %}
 
+{% macro get_relation_type(relation) -%}
+  {% set result_set = run_query('describe extended ' ~ relation) %}
+  {% set rel_type = 'table' %}
+  {%- for rs in result_set -%}
+    {%- if rs[0].startswith('Table Type') -%}
+      {%- if rs[1].startswith('VIRTUAL_VIEW') -%}
+        {% set rel_type = 'view' %}
+        {% do return(rel_type) %}
+      {%- elif rs[1].startswith('MANAGED_TABLE') -%}
```

Comment:
This should also look for 'EXTERNAL_TABLE' type for tables created with CREATE EXTERNAL TABLE

Reply:
@sdairs thanks for the catch, will modify.

Reply:
@sdairs force pushed a change for the above.